extensions/DynamicPageList/DPLSetup.php

```diff
@@ -1436,12 +1436,12 @@ public static function dplReplaceParserFunction(&$parser, $text, $pat, $repl='')
         return preg_replace ( $pat, $repl, $text );
     } 
 
-    public static function dplChapterParserFunction(&$parser, $text='', $heading=' ', $maxLength = -1, $page = '?page?', $link = 'default', $trim=false ) {
+    public static function dplChapterParserFunction($parser, $text='', $heading=' ', $maxLength = -1, $page = '?page?', $link = 'default', $trim=false ) {
         $output = DPLInclude::extractHeadingFromText($parser, $page, '?title?', $text, $heading, '', $sectionHeading, true, $maxLength, $link, $trim);
         return $output[0];
     } 
 
-    public static function dplMatrixParserFunction(&$parser, $name, $yes, $no, $flip, $matrix ) {
+    public static function dplMatrixParserFunction($parser, $name, $yes, $no, $flip, $matrix ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "dplMatrixParserFunction" is 24 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)